### PR TITLE
Add the posibility to move mailboxes

### PR DESCRIPTION
**Pull request by GretaD**
Signed-off-by: GretaD <gretadoci@gmail.com>

### CODE REVIEW

**Comment by ChristophWurst** [comment]
To bring this to life

* Update to latest master once https://github.com/nextcloud/mail/pull/3943 is in
* Update `nextcloud/vue` to 3.1
* Move the modal into the `#extra` slot of the app nav item

Then it will properly show for all mailboxes :)

**Comment by GretaD** [comment]
@ChristophWurst should i also rename the movemodal.vue into movemessagemodal.vue and include here or as a separate PR?

File: src/components/MailboxPicker.vue [hunk 1]
```diff
@@ -77,6 +77,14 @@ export default {
 			type: String,
 			default: t('mail', 'Choose'),
 		},
+		pickedMailbox: {
+			type: Object,
+			required: false,
```
Comment (ChristophWurst) [inline src/components/MailboxPicker.vue:82]: needs a default (as function) https://vuejs.org/v2/guide/components-props.html#Prop-Validation

**Comment by ChristophWurst** [comment]
Needs rebase

**Review by ChristophWurst — APPROVED** [review]

**Comment by ChristophWurst** [comment]
Minor detail I noticed while testing https://github.com/nextcloud/mail/issues/4045

**Pull request merged**